pk3DS.Core/Randomizers/PersonalRandomizer.cs

```diff
@@ -67,56 +75,116 @@ public void Execute()
         {
             for (var i = 1; i < Table.Length; i++)
                 Randomize(Table[i], i);
+            
+            List<Action<PersonalInfo, PersonalInfo>> propagations = new List<Action<PersonalInfo, PersonalInfo>>();
+            List<Action<PersonalInfo>> propagationDeviations = new List<Action<PersonalInfo>>();
 
             if (TMInheritance)
-                PropagateTMs(Table, Evos);
+            {
+                propagations.Add((a, b) =>
+                {
+                    for (var i = 0; i < a.TMHM.Length; i++)
+                    {
+                        b.TMHM[i] = a.TMHM[i];
+                    }
+                });
+                if (TMInheritanceDeviation > 0)
+                {
+                    propagationDeviations.Add(a =>
+                    {
+                        for (int i = 0; i < a.TMHM.Length; i++)
+                        {
+                            // Randomly flip whether TMs are learned TMInheritanceDeviation% of the time
+                            a.TMHM[i] ^= (rnd.Next(100) < TMInheritanceDeviation);
+                        }
+                    });
+                }
+            }
+
+
+            if (AbilityInheritance)
+            {
+                propagations.Add((a, b) => a.Abilities = b.Abilities);
```

Comment:
Might want a .ToList or .ToArray at the end of b.Abilities. Otherwise, any future changes to b.Abilities will also be made to a.Abilities since they share the same reference now.